### DISCUSSION TITLE
Add slots for image at left and right end of ToggleSwitch component.

### DISCUSSION THREAD

**Discussion by goldengecko** [primefaces/community#2201] [PrimeVue] [answered]
Could we please have slots there to make it easy to add icons behind the on and off state? 

My immediate use case is for a dark mode switch like this:
![Screenshot 2024-06-25 at 6 26 33 PM](https://github.com/primefaces/community/assets/703713/bdaf5616-cf67-49e9-a823-600dc70730d2)
![Screenshot 2024-06-25 at 6 26 50 PM](https://github.com/primefaces/community/assets/703713/083f8391-4ecd-4015-af82-102c745425dc)

There is a similar suggestion for a slot to be able to add an image on the thumb itself, so perhaps both options could be implemented at the same time to keep everyone happy.

**Comment by vieira-brz** [comment, marked as answer]
Here’s a simplified approach to achieve this with web components or frameworks that support slots like Vue, Angular, or Web Components (custom elements):

### Step 1: Define the Component Structure
If you're using a framework that supports slots, you can define your toggle switch component with slots for each state. Here’s a basic example using Web Components:

```html
<!-- Define the custom toggle switch element -->
<template id="toggle-switch-template">
  <style>
    :host {
      display: inline-block;
      position: relative;
      width: 50px; /* width of the switch */
      height: 25px; /* height of the switch */
    }
    #switch {
      position: absolute;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      background-color: #ccc;
      border-radius: 12.5px;
      cursor: pointer;
      transition: background 0.3s;
    }
    #thumb {
      position: absolute;
      top: 2.5px;
      bottom: 2.5px;
      left: 2.5px;
      width: 20px;
      background-color: white;
      border-radius: 50%;
      transition: transform 0.3s;
    }
    :host([checked]) #switch {
      background-color: #4CAF50;
    }
    :host([checked]) #thumb {
      transform: translateX(25px);
    }
  </style>
  <div id="switch">
    <div id="thumb"></div>
    <slot name="icon-off"></slot>
    <slot name="icon-on"></slot>
  </div>
</template>

<script>
  class ToggleSwitch extends HTMLElement {
    constructor() {
      super();
      this.attachShadow({mode: 'open'});
      this.shadowRoot.appendChild(document.getElementById('toggle-switch-template').content.cloneNode(true));
      this.addEventListener('click', this.toggle.bind(this));
    }

    connectedCallback() {
      this.setAttribute('aria-checked', this.checked.toString());
    }

    get checked() {
      return this.hasAttribute('checked');
    }

    set checked(val) {
      if (val) {
        this.setAttribute('checked', '');
      } else {
        this.removeAttribute('checked');
      }
      this.setAttribute('aria-checked', this.checked.toString());
    }

    toggle() {
      this.checked = !this.checked;
    }
  }

  window.customElements.define('toggle-switch', ToggleSwitch);
</script>
```

### Step 2: Use the Component with Custom Icons
Now, use the toggle switch in your HTML and specify custom icons for each slot:

```html
<toggle-switch>
  <img slot="icon-off" src="icon-moon.png" style="position:absolute; left:5px; top:5px;">
  <img slot="icon-on" src="icon-sun.png" style="position:absolute; right:5px; top:5px;">
</toggle-switch>
```

### Notes
- **Styling**: Adjust the CSS to properly place the icons or images. You might need to tweak positions based on the actual sizes of your icons.
- **Framework Integration**: If you’re using Vue or Angular, the concept of slots is similar but integrated differently. Vue uses `<slot>` directly, and Angular has `ng-content` for content projection.

**Reply by goldengecko** [reply]
Excellent, thanks for that.
